Misc/NEWS.d/next/Core and Builtins/2023-06-16-21-29-06.gh-issue-105858.Q7h0EV.rst

```diff
@@ -0,0 +1,7 @@
+Improve the constructors for :mod:`ast` nodes. Arguments of list types now
+default to an empty list if omitted. AST nodes now have an
+``__annotations__`` attribute with the expected types of their attributes.
+Passing unrecognized extra arguments to AST nodes is deprecated and will
+become an error in Python 3.15. Omitting a required argument to an AST node
+is deprecated and will become an error in Python 3.15. Patch by Jelle
```

Comment:
Is there any way to explore the likely impact of these deprecations in the wild?

The new behavior seems much more reasonable to me; less error-prone and more inline with the usual behavior of Python objects. But it still seems like we should weigh these clear benefits against some idea of the impact/churn on downstream code.

Has this been discussed anywhere public (e.g. on Discourse) to give interested parties a chance to speak up if they have code that would require lots of updates to adhere to these new requirements?

Reply:
Good point, opened https://discuss.python.org/t/proposed-change-to-ast-node-constructors/46531

Reply:
I think maybe this might be worth a short note in "What's new in Python 3.13" as well?

Reply:
Yes, I should also mention it in `ast.rst`.

Reply:
Just pushed some docs changes, let me know what you think.